packages/terra-functional-testing/src/services/wdio-visual-regression-service/methods/BaseCompare.js

```diff
@@ -1,27 +1,128 @@
 /* eslint-disable class-methods-use-this, no-unused-vars */
+import path from 'path';
+
+
+const TEST_ID_REGEX = /\[([^)]+)\]/;
 
 export default class BaseCompare {
+  constructor(options) {
+    const {
+      baseScreenshotDir,
+      locale,
+      theme,
+    } = options;
+
+    // screenshot naming config
+    this.baseScreenshotDir = baseScreenshotDir || process.cwd();
+    this.locale = locale || 'en';
+    this.theme = theme || 'terra-default-theme';
+  }
+
   /**
    * You can do here your image comparison magic.
+   * @param {Object} context - information provided to process the screenshot
```

Comment:
Nit: Is there a grammar standard we want to adopt here? Case and sentence endings are hit and miss.

>   * @param {Object} context - Information provided to process the screenshot.

Reply:
I'm good with a standard. Want to shoot for correct grammar with caps, punctuation, etc ?

Reply:
That works for me, I'm not too opinionated which way we go, just that it's consistent

Reply:
Thoughts? https://github.com/cerner/terra-toolkit/pull/458/commits/0550c7e5942ba7c51cbba913fa44dbc07abea494

Reply:
Works for me, I'm not a huge fan of the newline break between the descriptions and params, but I think a few people prefer this syntax and it's a personal preference.